### PR TITLE
MNT: Update infrastructure

### LINKED ISSUE #21: Add python_requires to setup.cfg
In https://github.com/astropy/pytest-astropy-header/pull/19, the Python 2.7 testing was dropped. We should explicitly define ``python_requires`` in ``setup.cfg`` to officially make 3.6 the oldest supported version.

cc @pllim

### PR DESCRIPTION
**Pull request by pllim**
Fix #21 

Supersedes #22 

cc @astrofrog 

This is like astropy/pytest-remotedata#48 + astropy/pytest-remotedata#49

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+changedir = .tmp/{envname}
+description = run tests
+deps =
+    astropy
```
Comment (pllim) [inline tox.ini]: @astrofrog , why didn't tox pick up deps from `setup.cfg`? Did I do this wrong?

Comment (astrofrog) [reply tox.ini]: I think we should include:

```
extras = test
```

to pull in the test dependencies from setup.cfg

**Review by astrofrog — APPROVED** [review]
Looks good! The test failure looks real due to a change in pytest dev? If so we should probably just merge this then open an issue about the failure.

**Comment by pllim** [comment]
Weird, why did it pass for 2nd commit?

**Comment by pllim** [comment]
I think it's the coverage stuff, that we don't need here anyway. I am removing them.

**Comment by pllim** [comment]
p.s. Also I cannot merge here, so it will have to be you, @astrofrog .

**Pull request merged**